kernel/comps/virtio/src/device/network/device.rs

```diff
@@ -170,29 +191,68 @@ fn queue_to_network_error(err: QueueError) -> VirtioNetError {
     match err {
         QueueError::NotReady => VirtioNetError::NotReady,
         QueueError::WrongToken => VirtioNetError::WrongToken,
+        QueueError::BufferTooSmall => VirtioNetError::Busy,
         _ => VirtioNetError::Unknown,
     }
 }
 
+fn init_caps(features: &NetworkFeatures, config: &VirtioNetConfig) -> DeviceCapabilities {
+    let mut caps = DeviceCapabilities::default();
+
+    caps.max_burst_size = None;
+    caps.medium = Medium::Ethernet;
+
+    if features.contains(NetworkFeatures::VIRTIO_NET_F_MTU) {
+        // If `VIRTIO_NET_F_MTU` is negotiated, the MTU is decided by the device.
+        caps.max_transmission_unit = config.mtu as usize;
+    } else {
+        // Linux does not has a specified MTU for virtio-net.
+        // Linux only requires the MTU between 68-65535.
+        // The exact value can be decided dynamically.
+        // <https://elixir.bootlin.com/linux/v6.0.9/source/drivers/net/virtio_net.c#L3660>.
+
+        if features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_TSO4)
+            || features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_TSO6)
+            || features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_UFO)
+        {
+            // If any of the above features is set, the max incoming packets is 65550 bytes
+            // caps.max_transmission_unit = 65550; 
+            todo!("support TSO4, TSO6 or UFO features");
+        } else {
+            // Else, the incoming packets is 1514 bytes.
+            caps.max_transmission_unit = 1514; 
+        }
+    }
+
+    // TODO: When we supporting more virtio-net features,
+    // the checksum may be offloaded to device instead.
+    // For transmitting: If `VIRTIO_NET_F_CSUM` is _not_ negotiated,
+    // the driver must supply a fully checksummed packet to the device.
+    // For receiving: If `VIRTIO_NET_F_GUEST_CSUM` is negotiated,
+    // the device has validated the packet checksum.
+    caps.checksum.tcp = Checksum::Both;
+    caps.checksum.udp = Checksum::Both;
+    caps.checksum.ipv4 = Checksum::Both;
+    caps.checksum.icmpv4 = Checksum::Both;
```

Comment:
The comments are not accurate. "must supply" and "has validated" are just false claims.

I would prefer something like this:
```rust
    // We do not support checksum offloading. So the features must not be negotiated, and we must
    // deliver fully checksummed packets to the device and validate all checksums for packets from
    // the device.
    assert!(!features.contains(NetworkFeatures::VIRTIO_NET_F_CSUM) &&
        !features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_CSUM));
    /* set csum caps... */
```